src/engine.py

```diff
@@ -21,37 +31,81 @@ def enqueue(self, path: str) -> bool:
         Returns:
             bool: Was enqueueing successfull
         """
-        raise NotImplementedError()
+        if not os.path.isdir(path):
+            return False
+        
+        folder_name = os.path.basename(os.path.normpath(path))
+        output_path = os.path.join(r"D:\processed_songs", folder_name)
+        os.makedirs(output_path, exist_ok=True)
+
+        task = self.manager.dict({
+            "is_done": False,
+            "has_failed": False,
+            "input_path": os.path.join(path, "audio.mp3"),
+            "output_path": output_path,
+            "expire_time": None
+        })
+
+        self.task_queue.append(task)
+        return True
+
+
     
-    def process_song(self) -> int:
+    def run(self):
         """
-        Processes a single song inside queue, handles switching to next song in queue
-
-        Returns:
-            status_code (int): 0: OK, 1: Failed
+        Main Engine loop, handles switching tasks and processing them in non-blocking way
         """
-        raise NotImplementedError()
+        print("Engine started.")
+        while True:
+            if len(self.task_queue) == 0:
+                time.sleep(1)
+                continue
+
+            task = self.task_queue.pop(0)
+            print(f"Processing: {task['input_path']}")
+            process = Process(target=Engine._process_wrapper, args=(task,))
+            process.start()
+            process.join()
```

Comment:
this line blocks the thread until we finish processing current task, so we can process only one task at a time if I understand correctly. Is this intended? If so I would remove the 'non-blocking' from comment above or elaborate